### PR TITLE
More megatron-enabled models

### PR DESCRIPTION
**Pull request by borisfom**
More megatron-enabled models. Also added parallel Jenkins test.

### CODE REVIEW

**Comment by borisfom** [comment]
Pending adding more Jenkins tests

File: nemo/collections/nlp/modules/common/megatron/megatron_utils.py [hunk 1]
```diff
@@ -82,16 +87,19 @@ def get_megatron_lm_model(pretrained_model_name: str, config_file: Optional[str]
                 "num_layers": configf['num-layers'],
                 "max_position_embeddings": configf['max-seq-length'],
             }
+    elif config_dict:
+        config = config_dict
     else:
         config = get_megatron_config(pretrained_model_name)
     if config is None:
         raise ValueError(f'Config file is required for {pretrained_model_name}')
```
Comment (ericharper) [inline nemo/collections/nlp/modules/common/megatron/megatron_utils.py:95]:
```suggestion
        raise ValueError(f'Config file or config dict is required for {pretrained_model_name}')
```

File: nemo/collections/nlp/modules/common/common_utils.py [hunk 2]
```diff
@@ -59,10 +59,11 @@ def get_pretrained_lm_model(
     else:
         if pretrained_model_name in get_megatron_lm_models_list():
             model, default_checkpoint_file = get_megatron_lm_model(
-                config_file=config_file, pretrained_model_name=pretrained_model_name
+                config_dict=config_dict,
+                config_file=config_file,
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/common_utils.py:63]: why do we need config_dict to Megatron? Megatron doesn't support it

File: nemo/collections/nlp/modules/common/megatron/megatron_utils.py [hunk 3]
```diff
@@ -82,16 +87,19 @@ def get_megatron_lm_model(pretrained_model_name: str, config_file: Optional[str]
                 "num_layers": configf['num-layers'],
                 "max_position_embeddings": configf['max-seq-length'],
             }
+    elif config_dict:
```
Comment (ekmb) [inline nemo/collections/nlp/modules/common/megatron/megatron_utils.py:90]: not supported, not tested

File: tests/collections/asr/test_asr_exportables.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import tempfile
+
+import pytest
+from omegaconf import DictConfig
+
+from nemo.collections.asr.modules import ConvASRDecoder, ConvASREncoder
+
+
+class TestExportable:
+    @pytest.mark.unit
```
Comment (ekmb) [inline tests/collections/asr/test_asr_exportables.py:24]: wasn't it the part of the yesterday's PR?

File: nemo/collections/nlp/modules/common/common_utils.py (hunk 2 above)
Comment (ericharper) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: I think we should be able to configure Megatron from yaml. I hate to force a user to pass in a .json file when we already have a yaml.

Comment (ekmb) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: do we have a working yaml megatron config? are you proposing to change all default model's configs to support this?

Comment (borisfom) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: No, I was just trying to unify interface for the two models. If I was to run all the tests written for HF with Megatron, I would like it to be able to take same arguments.

File: nemo/collections/nlp/modules/common/megatron/megatron_utils.py (hunk 3 above)
Comment (borisfom) [reply nemo/collections/nlp/modules/common/megatron/megatron_utils.py:90]: Not supported? How is that different from reading from config file ? We do pass a dict down to actual Megatron constructor.

File: tests/collections/asr/test_asr_exportables.py (hunk 4 above)
Comment (borisfom) [reply tests/collections/asr/test_asr_exportables.py:24]: Yes it was - not sure why it's coming up here, will check.

File: nemo/collections/nlp/modules/common/common_utils.py (hunk 2 above)
Comment (ekmb) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: what I'm unclear about is where does this separate yaml is coming from? for .json file - it is released on NGC along with the pretrained megatron checkpoint, default yaml model files won't contain megatron configs, so where the user is going to get it?

Comment (ericharper) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: @ekmb I have some ideas on how we can do this really nicely for NeMo users. But that is beyond the scope of this PR. For this PR, if we have config_dict option, then we can put megatron config into the yaml like we did for bert pretraining: https://github.com/NVIDIA/NeMo/blob/candidate/examples/nlp/language_modeling/conf/bert_pretraining_from_text_config.yaml

Comment (ericharper) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: Note that, we don't have to put anything into the yaml, and we can specify the .json file as the config_file has precedence. By having config_dict we have more flexibility and I don't think it's harming anything.

Comment (ekmb) [reply nemo/collections/nlp/modules/common/common_utils.py:63]: @ericharper we can't add megatron config to every model configs - this is just impractical. 
I'm fine with leaving this as-is, it just might confuse users imo

**Pull request closed without merging**